runtime/hydradx/src/evm/aave_trade_executor.rs

```diff
@@ -0,0 +1,400 @@
+		let data = Into::<u32>::into(Function::UnderlyingAssetAddress)
+			.to_be_bytes()
+			.to_vec();
+
+		let (res, value) = Executor::<T>::view(context, data, VIEW_GAS_LIMIT);
+
+		if !matches!(res, Succeed(_)) {
+			// not a token
+			return None;
+		}
+
+		Some(EvmAddress::from(H256::from_slice(&value)))
+	}
+
+	fn supply(origin: OriginFor<T>, asset: EvmAddress, amount: Balance) -> Result<(), DispatchError> {
+		let who = ensure_signed(origin)?;
+		let on_behalf_of = T::EvmAccounts::evm_address(&who);
+
+		let context = CallContext::new_call(<BorrowingContract<T>>::get(), on_behalf_of);
+		let data = EvmDataWriter::new_with_selector(Function::Supply)
+			.write(asset)
+			.write(amount)
+			.write(on_behalf_of)
+			.write(0_u16)
+			.build();
+
+		handle_result(Executor::<T>::call(context, data, U256::zero(), TRADE_GAS_LIMIT))
+	}
+
+	fn withdraw(origin: OriginFor<T>, asset: EvmAddress, amount: Balance) -> Result<(), DispatchError> {
+		let who = ensure_signed(origin)?;
+		let to = T::EvmAccounts::evm_address(&who);
+
+		let context = CallContext::new_call(<BorrowingContract<T>>::get(), to);
+		let data = EvmDataWriter::new_with_selector(Function::Withdraw)
+			.write(asset)
+			.write(amount)
+			.write(to)
+			.build();
+
+		handle_result(Executor::<T>::call(context, data, U256::zero(), TRADE_GAS_LIMIT))
+	}
+
+	pub fn trade_weight() -> Weight {
+		<T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(TRADE_GAS_LIMIT + VIEW_GAS_LIMIT, true)
+	}
+}
+
+fn handle_result(result: CallResult) -> DispatchResult {
+	let (exit_reason, value) = result;
+	match exit_reason {
+		Succeed(_) => Ok(()),
+		e => {
+			let hex_value = hex::encode(&value);
+			log::error!(target: "evm", "evm call failed with : {:?}, value: 0x{}, decoded: {:?}", e, hex_value, String::from_utf8_lossy(&value).into_owned());
+			Err(DispatchError::Other(&*Box::leak(
+				format!("evm:0x{}", hex_value).into_boxed_str(),
+			)))
+		}
+	}
+}
+
+impl<T> TradeExecution<OriginFor<T>, AccountId, AssetId, Balance> for AaveTradeExecutor<T>
+where
+	T: pallet_evm::Config
+		+ pallet_asset_registry::Config<AssetId = AssetId>
+		+ pallet_liquidation::Config
+		+ pallet_evm_accounts::Config
+		+ frame_system::Config,
+	T::AssetNativeLocation: Into<MultiLocation>,
+	BalanceOf<T>: TryFrom<U256> + Into<U256>,
+	T::AddressMapping: pallet_evm::AddressMapping<T::AccountId>,
+	<T as frame_system::Config>::AccountId: AsRef<[u8; 32]>,
+	pallet_evm::AccountIdOf<T>: From<T::AccountId>,
+	NonceIdOf<T>: Into<T::Nonce>,
+{
+	type Error = DispatchError;
+
+	fn calculate_sell(
+		pool_type: PoolType<AssetId>,
+		_asset_in: AssetId,
+		_asset_out: AssetId,
+		amount_in: Balance,
+	) -> Result<Balance, ExecutorError<Self::Error>> {
+		if pool_type != PoolType::Aave {
+			return Err(ExecutorError::NotSupported);
+		}
+
+		// For both supply and withdraw, amount out is always 1:1
+		// to save weight we just assume the operation will be available
+		Ok(amount_in)
```

Comment:
here we need to calculate amount_out, why do we just return amount_in?

Reply:
price is 1 always